### PR TITLE
pythonPackages.ansible-runner: skip failing test on Catalina

### PR DESCRIPTION
**Pull request by kalbasit**
###### Motivation for this change

Catalina replaced the director /home with a symbolic link to `/System/Volumes/Data/home` and this test is looking for /home explicitly.

I filed https://github.com/ansible/ansible-runner/issues/413 to get this fix upstream.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ansible-runner/default.nix [hunk 1]
```diff
@@ -30,8 +31,10 @@ buildPythonPackage rec {
     six
   ];
 
+  # test_process_isolation_settings is currently broken on Darwin Catalina
+  # https://github.com/ansible/ansible-runner/issues/413
   checkPhase = ''
-    HOME=$(mktemp -d) pytest --ignore test/unit/test_runner.py -k "not test_prepare"
+    HOME=$(mktemp -d) pytest --ignore test/unit/test_runner.py -k "not test_prepare${lib.optionalString stdenv.isDarwin " and not test_process_isolation_settings"}"
```
Comment (jonringer) [inline pkgs/development/python-modules/ansible-runner/default.nix]: all tests have to be prefixed with test_, so it's not very useful to identify tests. And I think i would prefer the whitespace to be around the entry
```suggestion
    HOME=$TMPDIR pytest \
      --ignore test/unit/test_runner.py \
      -k "not prepare ${lib.optionalString stdenv.isDarwin "and not process_isolation_settings"}"
```

Comment (kalbasit) [reply pkgs/development/python-modules/ansible-runner/default.nix]: ok trying your suggestion.

**Comment by kalbasit** [comment]
It seems to work fine! I'm planning to squash-merge the pr so no need to rebase commits.

@GrahamcOfBorg build python2Packages.ansible-runner python3Packages.ansible-runner

**Comment by jonringer** [comment]
otherwise LGTM

**Pull request merged**